recipes/cuda-cccl-impl/meta.yaml

```diff
@@ -0,0 +1,67 @@
+{% set name = "cuda-cccl-impl" %}
+{% set version = "2.0.1" %}
+{% set thrust_version = "2.0.1" %}
+{% set cub_version = "2.0.1" %}
+{% set libcudacxx_version = "1.9.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.gz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
```

Comment:
Maybe something like this can address @beckermr's request for documenting the relationships between packages. I would appreciate input from @adibbley @jakirkham on this.
```suggestion
# CUDA C++ Core Libraries (CCCL) includes thrust, cub, and libcudacxx.
# The cuda-cccl-impl package ships CCCL headers in the environment's include directory for use in downstream libraries. It follows CCCL upstream versioning.
# The cuda-cccl_{{ target_platform }} package (e.g. cuda-cccl_linux-64) ships CCCL headers in 'targets/.../include' that can be used by nvcc, the CUDA compiler. It follows CUDA Toolkit versioning.
# The cuda-cccl package is a metapackage that depends on cuda-cccl-impl and cuda-cccl_{{ target_platform }} to ensure that versions are consistent. That is, "I want to use the thrust headers that shipped with CUDA Toolkit 12.1."


package:
```

Reply:
Good start. You should cover why there are separate packages per target.

Reply:
Yeah the separate packages per target is needed due to cross-compilation (as this is a dependency of the compiler, `cuda-nvcc`)

Reply:
Edited to mention cross-compilation in the target_platform description.

Reply:
LGTM, thanks!